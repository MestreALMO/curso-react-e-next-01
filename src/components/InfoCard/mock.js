const props = { title: "Title", text: "Text" };

export default props;
