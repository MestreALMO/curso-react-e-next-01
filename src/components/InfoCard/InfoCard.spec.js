import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { InfoCard } from ".";
import mock from "./mock";

describe("<InfoCard />", () => {
  it("should render InfoCard correctly", () => {
    render(<InfoCard {...mock} />);

    expect(screen.getByRole("img", { ...mock })).toBeInTheDocument();
    expect(screen.getByRole("img", { ...mock })).toHaveAttribute(
      "src",
      require("../../images/thumbnails/styled-components.png")
    );
  });

  it("should match snapshot", () => {
    const { container } = render(<InfoCard />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
