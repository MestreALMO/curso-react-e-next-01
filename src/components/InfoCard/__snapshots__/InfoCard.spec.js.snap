// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoCard /> should match snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-15lq9tr-MuiPaper-root-MuiCard-root"
>
  <img
    src="styled-components.png"
    style="max-width: 400px;"
  />
  <div
    class="MuiBox-root css-0"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    />
  </div>
</div>
`;
